examples/js/server-rendering/instantsearch-server.js

```diff
@@ -0,0 +1,113 @@
+import http from 'node:http';
+import { JSDOM, ResourceLoader } from 'jsdom';
+import { readFile } from 'node:fs/promises';
+
+import {
+  getInitialResults,
+  waitForResults,
+} from 'instantsearch.js/es/lib/server.js';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import instantsearch from 'instantsearch.js/dist/instantsearch.development.js';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import algoliasearch from 'algoliasearch/dist/algoliasearch-lite.esm.browser.mjs';
+
+class CustomResourceLoader extends ResourceLoader {
+  fetch(url, options) {
+    if (options.element.name === 'HTMLLinkElement') {
+      return Promise.resolve(Buffer.from(''));
+    }
+
+    return super.fetch(url, options);
+  }
+}
+
+http
+  .createServer(async (req, res) => {
+    const clientURL = new URL(
+      req.url,
+      'http://localhost:3000'
+    ).searchParams.get('url');
+
+    const { window } = new JSDOM(
+      await readFile(
+        new URL('./instantsearch-code.html', import.meta.url),
+        'utf-8'
+      ),
+      {
+        url: new URL(clientURL, 'http://localhost:3000'),
+        resources: new CustomResourceLoader(),
+        runScripts: 'dangerously',
+      }
+    );
+
+    await new Promise((resolve) => {
+      window.addEventListener('load', resolve);
+    });
+
+    window.setTimeout = setTimeout;
+    window.setInterval = setInterval;
+
+    Object.assign(global, window, {
+      window,
+      // @TODO: how to add all global window properties?
+      HTMLElement: window.HTMLElement,
+      XMLHttpRequest: window.XMLHttpRequest,
+    });
```

Comment:
in a real use case, we wouldn't add these all in current global scope, but as exposed variables to the isolate which runs the instantsearch code.